cuda_core/cuda/core/experimental/_memory/_memory_pool.pyx

```diff
@@ -0,0 +1,438 @@
+                    &current_threshold
+                )
+            )
+
+            # If threshold is 0 (default), set it to maximum to retain memory in the pool.
+            if current_threshold == 0:
+                HANDLE_RETURN(cydriver.cuMemPoolSetAttribute(
+                    self._handle,
+                    cydriver.CUmemPool_attribute.CU_MEMPOOL_ATTR_RELEASE_THRESHOLD,
+                    &max_threshold
+                ))
+        elif opts._type == cydriver.CUmemAllocationType.CU_MEM_ALLOCATION_TYPE_PINNED \
+                and opts._location == cydriver.CUmemLocationType.CU_MEM_LOCATION_TYPE_HOST:
+            IF CUDA_CORE_BUILD_MAJOR >= 13:
+                assert dev_id == -1
+                loc.id = dev_id
+                loc.type = opts._location
+                HANDLE_RETURN(cydriver.cuMemGetMemPool(&(self._handle), &loc, opts._type))
+            ELSE:
+                raise RuntimeError("not supported")
+        elif opts._type == cydriver.CUmemAllocationType.CU_MEM_ALLOCATION_TYPE_PINNED \
+                and opts._location == cydriver.CUmemLocationType.CU_MEM_LOCATION_TYPE_HOST_NUMA:
+            IF CUDA_CORE_BUILD_MAJOR >= 13:
+                assert dev_id == 0
+                loc.id = 0
+                loc.type = opts._location
+                HANDLE_RETURN(cydriver.cuMemGetMemPool(&(self._handle), &loc, opts._type))
+            ELSE:
+                raise RuntimeError("not supported")
+        else:
+            IF CUDA_CORE_BUILD_MAJOR >= 13:
+                if opts._type == cydriver.CUmemAllocationType.CU_MEM_ALLOCATION_TYPE_MANAGED:
+                    # Managed memory pools
+                    loc.id = dev_id
+                    loc.type = opts._location
+                    HANDLE_RETURN(cydriver.cuMemGetMemPool(&(self._handle), &loc, opts._type))
+                else:
+                    assert False
+            ELSE:
+                assert False
+
+    return 0
+
+
+cdef int _MP_init_create(_MemPool self, int dev_id, _MemPoolOptions opts) except?-1:
+    cdef cydriver.CUmemPoolProps properties
+    memset(&properties, 0, sizeof(cydriver.CUmemPoolProps))
+
+    cdef bint ipc_enabled = opts._ipc_enabled
+    properties.allocType = opts._type
+    properties.handleTypes = _ipc.IPC_HANDLE_TYPE if ipc_enabled else cydriver.CUmemAllocationHandleType.CU_MEM_HANDLE_TYPE_NONE
+    properties.location.id = dev_id
+    properties.location.type = opts._location
+    # managed memory does not support maxSize as of CUDA 13.0
+    IF CUDA_CORE_BUILD_MAJOR >= 13:
+        if properties.allocType != cydriver.CUmemAllocationType.CU_MEM_ALLOCATION_TYPE_MANAGED:
+            properties.maxSize = opts._max_size
+    ELSE:
+        properties.maxSize = opts._max_size
+
+    self._dev_id = dev_id
+    self._mempool_owned = True
+
+    with nogil:
+        HANDLE_RETURN(cydriver.cuMemPoolCreate(&(self._handle), &properties))
+        # TODO: should we also set the threshold here?
```

Comment:
Just to confirm, we're currently setting the threshold higher up still, correct?

Reply:
No. Currently (before/after this PR), we only crank up the threshold when getting the current mempool, not when creating a new mempool. 

I think this is an oversight but I will address this in a follow-up PR. `_memory_pool.pyx` in this PR just a copy-pasta, not new code. Also, for new mempool instances I am not sure if setting threshold to the maximum is the safest thing to do. I'll play around with it but I suspect this won't be done by ~tomorrow.